mobile/src/main/java/com/joaquimley/avenging/ui/list/ListFragment.java

```diff
@@ -316,6 +318,12 @@ public boolean onQueryTextChange(String queryText) {
     }
 
     @Override
+    public void onDestroyView() {
+        mCharactersRecycler.setAdapter(null);
```

Comment:
Actually, at first I tried that but it didn't fix the leak. I also added detachment of adapter from RecyclerView, the memory leak went away. Then I removed detaching listener in adapter and the detachment of adapter from RecyclerView by itself was able to solve the leak.